subprojects/core/src/main/java/org/gradle/api/internal/classpath/DefaultModuleRegistry.java

```diff
@@ -196,38 +196,48 @@ private Module module(String moduleName, Properties properties, Set<File> implem
     }
 
     private void findImplementationClasspath(String name, Collection<File> implementationClasspath) {
-        List<String> suffixes = getClasspathSuffixes(name);
+        Matcher matcher = Pattern.compile("gradle-(.+)").matcher(name);
+        matcher.matches();
```

Comment:
This code was there before, I just moved it around when I split up the getSuffix method. Since it worked all the time, I think I would leave it as is.

Reply:
Yeah, I saw that it was there before. If the matcher doesn't match, then the next line throws an exception, right?